lib/modules/platform/gerrit/index.ts

```diff
@@ -0,0 +1,544 @@
+    await findOwnPr(
+      {
+        branchName: prConfig.sourceBranch,
+        targetBranch: prConfig.targetBranch,
+        state: 'open',
+      },
+      true
+    )
+  ).pop();
+  if (pr) {
+    //Workaround for "Known Problems.1"
+    if (pr.subject !== prConfig.prTitle) {
+      await updatePullRequestTitle(pr._number, pr.change_id, prConfig.prTitle);
+    }
+    await updatePullRequestBody(pr._number, prConfig.prBody);
+    if (prConfig.platformOptions?.gerritAutoApprove) {
+      await client.approveChange(pr._number);
+    }
+    return getPr(pr._number);
+  } else {
+    throw new Error(
+      `the change should be created automatically from previous push to refs/for/${prConfig.sourceBranch}`
+    );
+  }
+}
+
+async function updatePullRequestTitle(
+  number: number,
+  gerritChangeID: string,
+  prTitle: string
+): Promise<void> {
+  try {
+    await client.setCommitMessage(
+      number,
+      `${prTitle}\n\nChange-Id: ${gerritChangeID}\n`
+    );
+  } catch (err) {
+    logger.error(
+      { err },
+      `Can't set pull-request-title ${prTitle} as commit-msg for change ${gerritChangeID}/${number}`
+    );
+  }
+}
+
+async function updatePullRequestBody(
+  changeId: number,
+  prBody: string
+): Promise<void> {
+  const prBodyExists = await checkForExistingMessage(
+    changeId,
+    prBody,
+    TAG_PULL_REQUEST_BODY
+  );
+  if (!prBodyExists) {
+    await client.addMessage(changeId, prBody, TAG_PULL_REQUEST_BODY);
+  }
+}
+
+async function checkForExistingMessage(
+  changeId: number,
+  newMessage: string,
+  msgType: string | null
+): Promise<boolean> {
+  const newMsg = newMessage.trim(); //the last \n was removed from gerrit after the comment was added...
+  const messages = await client.getMessages(changeId);
+  return (
+    messages.find(
+      (existingMsg) =>
+        (msgType === null || msgType === existingMsg.tag) &&
+        existingMsg.message.includes(newMsg)
+    ) !== undefined
+  );
+}
+
+export async function getBranchPr(branchName: string): Promise<Pr | null> {
+  const change = (await findOwnPr({ branchName, state: 'open' })).pop();
+  if (change) {
+    return mapGerritChangeToPr(change);
+  }
+  return null;
+}
+
+export function getPrList(): Promise<Pr[]> {
+  return findOwnPr({ branchName: '' }).then((res) =>
+    res.map((change) => mapGerritChangeToPr(change))
+  );
+}
+
+export async function mergePr(config: MergePRConfig): Promise<boolean> {
+  logger.debug(
+    `mergePr(${config.id}, ${config.branchName!}, ${config.strategy!})`
+  );
+  try {
+    const change = await client.submitChange(config.id);
+    return change.status === 'MERGED';
+  } catch (err) {
+    if (err.statusCode === 409) {
+      logger.warn(
+        { err },
+        "Can't submit the change, because the submit rule doesn't allow it."
+      );
+      return false;
+    }
+    throw err;
+  }
+}
+
+/**
+ * BranchStatus for Gerrit: TODO: what should we check here? How can this work with: automergeType: "branch"
```

Comment:
normally we ready status values from ci runs. how does any CI report status to gerrit?

If that isn't supported we should always return yellow to not automerge without waiting for CI checks.
This can be overridden with `ignoreTests`[^1]

[^1]: https://docs.renovatebot.com/configuration-options/#ignoretests

Reply:
the CI reports the status to the Change (PR) as one or more label. From these labels the "submittable" status of the Change(PR) is determined.

Because there are no branches needed/used in Gerrit for generating PR's the `getBranchStatus`  function assumes that the `branchName` parameter refers to a PR.

If renovate was configured to `automergeType: "branch"` it will not try to create PR's. 
But for gerrit these PRs (Changes) are created automatically on pushing a commit. 

And therefore these `automergeType=branch` mode works (almost) exactly the same as the (default) "pr" mode for the Gerrit platform. 
"almost" because renovate at first create only a "branch" but after the push to Gerrit a change will be created automatically and on the next run renovate found these change and the behavior switches now implicit to "pr" mode.
Adding the PR-body (as a Gerrit message) and auto-Approve labels are delayed until this second run.

Therefore i would not suggest to use these `automergeType: "branch"` mode for Gerrit. 
There is zero benefit but some things are delayed.